### PR TITLE
Chapter 6: Modeling Users

### PR DESCRIPTION
**Pull request by puteriffah**

### CODE REVIEW

**Comment by puteriffah** [comment]
ready

File: config/application.rb [hunk 1]
```diff
@@ -10,4 +10,4 @@ class Application < Rails::Application
     config.i18n.available_locales = [:en, :ru]
     config.i18n.default_locale = :en
   end
-end
+end
```
Comment (ngocvt-0484) [inline config/application.rb]:
```suggestion
end

```

Comment (ngocvt-0484) [reply config/application.rb]: add new line at the botton of file
just only need enter

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,17 @@
...
+    length: { maximum: Settings.validations.email.max_length},
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
+
+  has_secure_password
+
+  before_save { email.downcase! }
```
Comment (ngocvt-0484) [inline app/models/user.rb]:
```suggestion
  before_save :downcase_email
  def downcase_email
     email.downcase!
  end
```

**Comment by puteriffah** [comment]
ready

File: app/models/user.rb [hunk 3]
```diff
@@ -1,2 +1,19 @@
 class User < ApplicationRecord
-end
+
```
Comment (saobv-2223) [inline app/models/user.rb]: delete empty line

File: app/models/user.rb [hunk 4]
```diff
@@ -1,2 +1,19 @@
...
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
+
+  has_secure_password
+
+  before_save :downcase_email
+  def downcase_email
```
Comment (saobv-2223) [inline app/models/user.rb]:
```suggestion
  before_save :downcase_email
  
  private 
  
  def downcase_email
```

**Comment by puteriffah** [comment]
ready

**Comment by ngocvt-0484** [comment]
good job.
merged

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**